### ISSUE TITLE
Selection of analysis results collections in collections table leads to unexpected result

### ISSUE THREAD

**Issue by fedorov** [ImagingDataCommons/IDC-WebApp#740]
If I click on a collection that is an analysis collection in the collections page, I am brought to the explorer page with URL populated with a filter, but filter blank.

Reated to this we should probably disable the ability to click on the analysis collections in the collections list until #590 is addressed.

**Comment by s-paquette** [comment]
@fedorov They're actually intended to be disabled, not sure why they aren't. I'll see if I can sneak in this fix for production.

**Comment by s-paquette** [comment]
So the issue here has to do with how we were determining the collections to select when an analysis result was clicked on. I've put in a fix for now, but I'm going to make a ticket about how this should function longer-term.

**Comment by fedorov** [comment]
I think long term, once there's progress for #590, clicking analysis results collection in the collections list should result in selecting that collection as a filter in the list of analysis results collections in the portal.

**Comment by s-paquette** [comment]
@fedorov Precisely.

**Comment by fedorov** [comment]
Test does not pass for me. Sometime it results in selection of collections (presumably, those that are touched by the analysis results collection?), and sometimes it does not do anything (such as for PROSTATEx-Seg-HiRes - although the filter is populated in the URL: https://testing-portal.canceridc.dev/explore/?filters_for_load=%5B%7B%22filters%22:%5B%7B%22id%22:%22120%22,%22values%22:%5B%22spie_aapm_nci%20prostatex%20challenges%20(prostate_x_challenge)%22%5D%7D%5D%7D%5D). What is the expected behavior?

**Comment by fedorov** [comment]
"Clear cache and hard reload" did not help.

**Comment by fedorov** [comment]
I will follow up with a query how to match these items to the original collection ids.

**Comment by s-paquette** [comment]
Okay, found these bugs, will flag this and tag both of you once the test build is up.

**Comment by fedorov** [comment]
@bcli4d I think you should use something like the below to get the list of collections "touched" by an analysis results collection instead of relying on scraping the wiki:

```sql
WITH
  analyses_results_studies AS (
  SELECT
    DISTINCT(StudyInstanceUID) AS StudyInstanceUID,
    analyses.DOI,
    analyses.Title
  FROM
    `canceridc-data.idc_v4.dicom_all` AS dicom_all
  JOIN
    `canceridc-data.idc_v4.analysis_results_metadata` AS analyses
  ON
    dicom_all.Source_DOI = analyses.DOI)
SELECT
  analyses_results_studies.DOI AS analysis_collection_DOI,
  STRING_AGG(DISTINCT(analyses_results_studies.Title)) as Title,
  STRING_AGG(DISTINCT(collection_id))
FROM
  analyses_results_studies
JOIN
  `canceridc-data.idc_v4.dicom_all` AS dicom_all
ON
  dicom_all.StudyInstanceUID = analyses_results_studies.StudyInstanceUID
GROUP BY
  analysis_collection_DOI
```

![image](https://user-images.githubusercontent.com/313942/135139403-a4f5533f-806d-4d5d-8ca7-77f98945e021.png)

**Comment by bcli4d** [comment]
Thanks for the SQL @fedorov. OK to wait for the next version, or does the analysis_results_metadata table get its own hotfix?

**Comment by s-paquette** [comment]
@pgundluru @fedorov Ready for testing.

**Comment by fedorov** [comment]
@bcli4d would it be possible to fix the table manually, and integrate something like the query above into ETL for the next release? It does bother me that the analysis results metadata table is not what user would expect it to contain.

@s-paquette it still does not work for me if I pick RIDER-LungCT-Seg.

**Comment by bcli4d** [comment]
I'll update the table tomorrow.

**Comment by bcli4d** [comment]
I'm not sure how these collection IDs are intended to be used. The collection ids returned by the above script are from dicom_all, which are "idc_webapp_collection_id"s. Is that what you need @s-paquette ? Those IDs are the "tcia_api_collections_id"s, with both spaces and dashes replaced by underscores, and lowered.

**Comment by bcli4d** [comment]
@s-paquette:..and should the Collections value be a single string of comma separated collection ids? or an array of strings?

**Comment by bcli4d** [comment]
I went ahead and update d`idc-dev-etl.idc_v4.analysis_results_metadata`. I've not pushed that to the public dataset. I'll do that when dev version is accepted.

**Comment by fedorov** [comment]
You meant - when the test version is accepted?

**Comment by bcli4d** [comment]
When the changes to idc-dev-etl.idc_v4.analysis_results_metadata are accepted.

**Comment by bcli4d** [comment]
idc-dev-etl.idc_v4.analysis_results_metadata copied to canceridc-data.idc_v4.analysis_results_metadata

**Comment by s-paquette** [comment]
@fedorov @pgundluru Hotfix applied to production (NOT in test or dev)--check when you have a moment, please!

**Comment by pgundluru** [comment]
Thank you Suzanne, Please see my results below:

DICOM-LIDC-IDRI - Success 
DICOM-SR-Breast-Clinical  - Fails with no errors in console 
PROSTATEx-Seg-HiRes - Fails 
PROSTATEx-Seg-Zones - Success 
QIN-LungCT-Seg - Fails 
RIDER-LungCT-Seg - Success

**Comment by s-paquette** [comment]
@pgundluru I can't reproduce a failure on PROSTATEx-Seg-HiRes - can you try it again? Found the problem and fixed the other two--try again when you get a chance, please.

**Comment by pgundluru** [comment]
Yes, Retested and i see that sometimes i will need to refresh once to get the filters to show up 

DICOM-LIDC-IDRI - Success
DICOM-SR-Breast-Clinical - Success 
PROSTATEx-Seg-HiRes - Success
PROSTATEx-Seg-Zones - Success
QIN-LungCT-Seg - Success 
RIDER-LungCT-Seg - Success
